src/libgraphviz/lib.rs

```diff
@@ -654,16 +728,26 @@ mod tests {
                     => lbls.into_iter().collect(),
             }
         }
+
+        fn len(&self) -> usize {
+            match self {
+                &UnlabelledNodes(len) => len,
+                &AllNodesLabelled(ref lbls) => lbls.len(),
+                &SomeNodesLabelled(ref lbls) => lbls.len(),
+            }
+        }
     }
 
     impl LabelledGraph {
         fn new(name: &'static str,
                node_labels: Trivial,
-               edges: Vec<Edge>) -> LabelledGraph {
+               edges: Vec<Edge>,
+               node_styles: Vec<Style>) -> LabelledGraph {
```

Comment:
could you make the `node_styles` an `Option<Vec<Style>>` instead?  In your tests, it seemed annoying to construct the vecs of no-style so frequently.